modules/xml/src/smithy4s/xml/Xml.scala

```diff
@@ -0,0 +1,126 @@
+/*
+ *  Copyright 2021-2022 Disney Streaming
+ *
+ *  Licensed under the Tomorrow Open Source Technology License, Version 1.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *     https://disneystreaming.github.io/TOST-1.0.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package smithy4s.xml
+
+import smithy4s.Blob
+import smithy4s.schema.Schema
+import fs2._
+import fs2.data.xml._
+import fs2.data.xml.dom._
+import cats.effect.SyncIO
+import cats.syntax.all._
+
+object Xml {
+
+  private val decoderCacheGlobal = XmlDocument.Decoder.createCache()
+  private val encoderCacheGlobal = XmlDocument.Encoder.createCache()
+
+  private def deriveXmlDecoder[A: Schema]: XmlDocument.Decoder[A] =
+    XmlDocument.Decoder.fromSchema(Schema[A], decoderCacheGlobal)
+
+  private def deriveXmlEncoder[A: Schema]: XmlDocument.Encoder[A] =
+    XmlDocument.Encoder.fromSchema(Schema[A], encoderCacheGlobal)
```

Comment:
better to have public members first, private members afterwards for people who like to read sources